dart/constraint/ConstraintSolver.h

```diff
@@ -110,11 +110,17 @@ class ConstraintSolver
 
   /// Set collision detector
   void setCollisionDetector(
-    std::unique_ptr<collision::CollisionDetector>&& _collisionDetector);
+    std::unique_ptr<collision::CollisionDetector> _collisionDetector);
 
   /// Get collision detector
   collision::CollisionDetector* getCollisionDetector() const;
 
+  /// Set LCP solver
+  void setLCPSolver(std::unique_ptr<LCPSolver> _lcpSolver);
```

Comment:
I see, thanks for the clarification. Good to know!

Also, this pull request looks good to merge.